File: data-azure-cosmos/src/test/java/io/micronaut/data/azure/entities/Child.java
package io.micronaut.data.azure.entities;

import io.micronaut.data.annotation.Relation;
import io.micronaut.serde.annotation.Serdeable;

import java.util.ArrayList;
import java.util.List;

@Serdeable
public class Child extends GenderAware {

    private String firstName;

    private int grade;

    @Relation(value = Relation.Kind.ONE_TO_MANY)
    private List<Pet> pets = new ArrayList<>();

    public String getFirstName() {
        return firstName;
    }

    public void setFirstName(String firstName) {
        this.firstName = firstName;
    }

    public int getGrade() {
        return grade;
    }

    public void setGrade(int grade) {
        this.grade = grade;
    }

    public List<Pet> getPets() {
        return pets;
    }

    public void setPets(List<Pet> pets) {
        this.pets = pets;
    }
}

File: doc-examples/azure-cosmos-example-kotlin/src/main/kotlin/example/Child.kt
package example

import io.micronaut.data.annotation.Relation
import io.micronaut.serde.annotation.Serdeable
import java.util.ArrayList

interface IGenderAware {
    var gender: String
}

data class GenderAware (
    override var gender: String
) : IGenderAware

@Serdeable
data class Child (
    var firstName: String,
    override var gender: String,
    var grade: Int,
    @Relation(value = Relation.Kind.ONE_TO_MANY)
    var pets: List<Pet> = ArrayList()
) : IGenderAware
